### PR TITLE
Shipping Classes: Create add/edit modal functionality

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/40190

This PR creates a modal to add and edit Shipping Classes while removing the previous in-line editing. Note the modal and table styles will be updated separately because the Shipping Zone Method modals and tables will also receive the same treatment, see https://github.com/woocommerce/woocommerce/issues/40472 and https://github.com/woocommerce/woocommerce/issues/40189.

image

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Note that we are testing only functionality in this PR, styles will be updated in a follow up.

1. Create a Jurassic Ninja site from this PR
2. Skip the setup but be sure to use a US address and USD (for test the Fedex extension in step 5)
3. Head to Shipping Classes, `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=classes`
4. Add and edit a Shipping Class. Test functionality by having a play with it to make sure data is correctly updated, persists on refresh, and deletes appropriately.
5. This PR reconfigures HTML derived from actions used by plugins to insert HTML into the shipping classes table. Test this by adding and activating such a plugin. For example, https://github.com/woocommerce/woocommerce-shipping-fedex
6. If you're testing with Fedex extension, go to `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=fedex` and click `Enable freight` then save.
7. Repeat step 4 testing, but this time note the extra column in the table and additional field in the modal.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add modal for adding and editing Shipping Classes

#### Comment 

</details>

### CODE REVIEW

**Review by alopezari — COMMENTED** [review]
Thanks for updating the E2E tests to cover this new change, @psealock!

These look good to me 👍

File: plugins/woocommerce/client/legacy/js/admin/wc-shipping-classes.js [hunk 1]
```diff
@@ -123,114 +92,112 @@
...
 					$tr.find( '.wc-shipping-class-delete' ).on( 'click', { view: this }, this.onDeleteRow );
-					$tr.find( '.editing .wc-shipping-class-edit' ).trigger('click');
-					$tr.find( '.wc-shipping-class-cancel-edit' ).on( 'click', { view: this }, this.onCancelEditRow );
-
-					// Editing?
-					if ( true === rowData.editing ) {
-						$tr.addClass( 'editing' );
-						$tr.find( '.wc-shipping-class-edit' ).trigger( 'click' );
-					}
+					// $tr.find( '.wc-shipping-class-cancel-edit' ).on( 'click', { view: this }, this.onCancelEditRow );
```
Comment (chihsuan) [inline plugins/woocommerce/client/legacy/js/admin/wc-shipping-classes.js:95]: Is this expected?

**Review by chihsuan — APPROVED** [review]
Nice work! Looks good and tested well. 🚢

File: plugins/woocommerce/client/legacy/js/admin/wc-shipping-classes.js (hunk 1 above)
Comment (psealock) [reply plugins/woocommerce/client/legacy/js/admin/wc-shipping-classes.js:95]: Oops, no I will remove this

**Pull request merged**